internal/common/conversion/patch_framework.go

```diff
@@ -0,0 +1,22 @@
+package conversion
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+func PatchPayloadHasChangesTpf[TFModel any, SDKRequest any](ctx context.Context, diags *diag.Diagnostics, state, plan *TFModel, converter func(ctx context.Context, input *TFModel, diags *diag.Diagnostics) *SDKRequest, reqPatch *SDKRequest) bool {
```

Comment:
what about reqPatch *SDKRequest being the returned value instead of bool, nil meaning there are no changes or an  errors was added to diagnostics? or it's because you can't create in in the func using like reqPatch = new(SDKRequest) ?